src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,337 @@
+
+  It requires correspondence configuration to function well.
+  The configuration key used is SignpostingConf.
+  It is a json structure shown below
+
+  useDefaultFileType is an on/off switch during linkset creating time, it controls whether the default type is
+  used, which is always Dataset
+
+  The configuration can be modified during run time by the administrator.
+ */
+
+import edu.harvard.iq.dataverse.Dataset;
+import edu.harvard.iq.dataverse.DataFile;
+import edu.harvard.iq.dataverse.DatasetAuthor;
+import edu.harvard.iq.dataverse.DatasetVersion;
+import edu.harvard.iq.dataverse.FileMetadata;
+import edu.harvard.iq.dataverse.dataaccess.StorageIO;
+import edu.harvard.iq.dataverse.dataaccess.SwiftAccessIO;
+import edu.harvard.iq.dataverse.license.License;
+
+import javax.json.Json;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import static edu.harvard.iq.dataverse.util.json.NullSafeJsonBuilder.jsonObjectBuilder;
+
+public class SignpostingResources {
+    private static final Logger logger = Logger.getLogger(SignpostingResources.class.getCanonicalName());
+    SystemConfig systemConfig;
+    DatasetVersion workingDatasetVersion;
+//    JsonObject licJsonObj;
+    JsonObject describedByJsonObj;
+    Boolean useDefaultFileType;
+    String defaultFileTypeValue;
+    int maxAuthors;
+    int maxItems;
+
+    public SignpostingResources(SystemConfig systemConfig, DatasetVersion workingDatasetVersion, String jsonSetting) {
+        this.systemConfig = systemConfig;
+        this.workingDatasetVersion = workingDatasetVersion;
+        if (jsonSetting == null) {
+            jsonSetting = BundleUtil.getStringFromBundle("signposting.configuration.SignpostingConf");
+        }
+        JsonReader jsonReader = Json.createReader(new StringReader(jsonSetting));
+        JsonObject spJsonSetting = jsonReader.readObject();
+        jsonReader.close();
+//        licJsonObj = spJsonSetting.getJsonObject("license");
+        describedByJsonObj = spJsonSetting.getJsonObject("describedby");
+        useDefaultFileType = spJsonSetting.getBoolean("useDefaultFileType", true);
+        defaultFileTypeValue = spJsonSetting.getString("defaultFileTypeValue", "https://schema.org/Dataset");
+        maxAuthors = spJsonSetting.getInt("maxAuthors", 5);
+        maxItems = spJsonSetting.getInt("maxItems", 5);
+    }
+
+    /**
+     * Get identifier schema for each author
+     * <p>
+     * For example:
+     * if author has VIAF
+     * Link: <http://viaf.org/viaf/:id/>; rel="author"
+     *
+     * @param datasetAuthors list of all DatasetAuthor object
+     * @return all the non empty author links in a string
+     */
+    private String getAuthors(List<DatasetAuthor> datasetAuthors) {
```

Comment:
sorry for the bad naming, renamed it to getAuthorsIdSchema, the purpose is to get the author Id schema from list of authors. Please see if the modification is fine, if not, please let me know which system call I can use instead. Thanks!

Reply:
I tried this approach, call and get json instead of using DatasetAuthors directly. It does not simplify the code, actually did just the opposite. 

The Json structure has to be converted anyway to string, while the resulting string is not a json string. The empty link check is redundant comparing to the json call, however, the logic is much simpler using datasetauthors rather than push every author into array, and then read them back, and then put into link string format.